src/metax_api/models/catalog_record_v2.py

```diff
@@ -122,6 +121,11 @@ def _post_create_operations(self, pid_type=None):
 
             self.research_dataset['preferred_identifier'] = 'draft:%s' % self.identifier
 
+            if self._get_preferred_identifier_type_from_request() == IdentifierType.DOI:
+                self.use_doi_for_published = True
+            else:
+                self.use_doi_for_published = False
+
```

Comment:
this is change is not saved because we're in post_create so save function have to be called separately. (should work when add the use_doi_for_published in save method below)